osidb/serializer.py

```diff
@@ -1359,6 +1362,95 @@ class Meta:
             + WorkflowModelSerializer.Meta.fields
         )
 
+    def update(self, new_flaw, validated_data):
+        """
+        perform the flaw instance update
+        with any necessary extra actions
+        """
+        #########################
+        # 1) pre-update actions #
+        #########################
+
+        # store the old flaw for the later comparison
+        old_flaw = Flaw.objects.get(uuid=new_flaw.uuid)
+
+        #####################
+        # 2) update actions #
+        #####################
+
+        # perform regular flaw update
+        new_flaw = super().update(new_flaw, validated_data)
+
+        ##########################
+        # 3) post-update actions #
+        ##########################
+
+        # update trackers if needed
+        self.update_trackers(old_flaw, new_flaw)
+
+        #####################
+        # 4) return updated #
+        #####################
+
+        return new_flaw
+
+    def update_trackers(self, old_flaw, new_flaw):
+        """
+        update the related trackers if needed
+        """
+        # this could be turned into a general
+        # helper if it was needed also elsewhere
+        def differs(instance1, instance2, attributes):
+            """
+            boolean check whether the given instances
+            have any differences in the given attributes
+
+            the caller is responsible for making sure that
+            the given instances really have the attributes
+            """
+            for attribute in attributes:
+                if getattr(instance1, attribute) != getattr(instance2, attribute):
+                    return True
+            return False
+
+        # TODO
+        # unembargo is more complicated
+        # and will be covered separately
+
+        # we only need to sync the trackers when crucial attributes change
+        # plus in the case of the impact we only care for the increase
+        #
+        # the crucial attributes are those influencing the SLA deadline plus the CVE ID
+        if not differs(
+            old_flaw, new_flaw, ["cve_id", "is_major_incident", "unembargo_dt"]
```

Comment:
`is_major_incident` is deprecated, so you should check `major_incident_state` (I think that change from `REQUESTED` to `APPROVED` in test should be triggerring)